build.sbt

```diff
@@ -864,6 +881,25 @@ lazy val testsCommonSettings = Def.settings(
   )
 )
 
+lazy val javaVersionBasedTestSources = Def.settings(
+  Test / unmanagedSourceDirectories ++= {
+    // unit-tests/native/src/test
+    val testDir = (Test / baseDirectory).value
+    sLog.value.info(s"testDir: $testDir")
+    val sharedTestDir =
+      testDir.getParentFile / "shared/src/test"
```

Comment:
https://github.com/scala-native/scala-native/blob/c3f13c43599b92da102b8883405a5311fca5b6d0/build.sbt#L834 code which does use glob filters to include all Scala files from `shared/src/test`

Reply:
Because of that `allScalaFromDir` then this works differently because Scala.js doesn't have that. Unless there is an easy way to change, I think we need to go with a non-standard setup.

Reply:
This is what I was missing, why it behaved not as expected.